### PR TITLE
updates

### PR DESCRIPTION
**Pull request by WindzCUHK**
1. export new configs as duration
    - add error `func New(opts ...Option) (Processor, error)`
1. add base claim to remove duplication

## TODO:
1. fix unit test
1. add new param in readme

### CODE REVIEW

File: option.go [hunk 1]
```diff
@@ -235,29 +238,29 @@ func WithJwkErrRetryInterval(i string) Option {
 }
 
 /*
-	processor parameters
+	access token parameters
 */
 
-// WithEnableMTLSCertificateBoundAccessToken returns a EnableMTLSCertificateBoundAccessToken functional option
-func WithEnableMTLSCertificateBoundAccessToken(b bool) Option {
+// WithATEnableMTLSCertificateBoundAccessToken returns a EnableMTLSCertificateBoundAccessToken functional option
```
Comment (kyfujisa) [inline option.go:244]: why did you add prefix "AT" ?
It is already in option name as "AccessToken".

File: option.go [hunk 2]
```diff
@@ -235,29 +238,29 @@ func WithJwkErrRetryInterval(i string) Option {
...
 	return func(authz *authorizer) error {
 		authz.enableMTLSCertificateBoundAccessToken = b
 		return nil
 	}
 }
 
-// WithProcessorClientCertificateGoBackSeconds returns a ProcessorClientCertificateGoBackSeconds functional option
-func WithProcessorClientCertificateGoBackSeconds(s int64) Option {
+// WithATProcessorClientCertificateGoBackSeconds returns a ProcessorClientCertificateGoBackSeconds functional option
+func WithATProcessorClientCertificateGoBackSeconds(t string) Option {
```
Comment (kevindiu) [inline option.go:253]: What is the difference between go back seconds and offset seconds? Can you write some comments?

Comment (kyfujisa) [reply option.go:253]: I wrote this part, so I'll answer that.

"GoBackSeconds" is the time you rewinded when issuing the certificate.
If there is a time delay due to a failure of NTP synchronization or some other reason, the certificate may be issued ahead of time to prevent "Not Before" from becoming the future.
In this case, go back seconds specifies the time of rewind.

"OffsetSeconds" is used to allow time for the CNF to accept failed verification.
During this time, even if the cnf fails to validate, if the "CommonName" of the certificate and the "client_id" of the token are the same, the validation will succeed.

comment is here.
https://github.com/yahoojapan/athenz-authorizer/pull/56/files#diff-3b639ccd4ab5dbda42bf990e4e9723f7R46

Comment (WindzCUHK) [reply option.go:253]: > What is the difference between go back seconds and offset seconds? Can you write some comments?
will update readme

**Comment by kyfujisa** [comment]
masterブランチへのPRではないので、マージします。CIのエラーなどは私が対応します。

**Comment by kyfujisa** [comment]
It's not a PR to the master branch, so I'll merge it, and I'll take care of the CI errors, etc.
